File: parking_lot_system_go/parkinglot/parking_level.go
package parkinglot

type ParkingLevel struct {
	LevelID int
	Spots   []*ParkingSpot
}

func NewParkingLevel(levelID, numSpots int) *ParkingLevel {
	level := &ParkingLevel{
		LevelID: levelID,
	}
	for i := 0; i < numSpots; i++ {
		var spotType VehicleType
		switch i % 3 {
		case 0:
			spotType = VehicleTypeCar
		case 1:
			spotType = VehicleTypeMotorcycle
		case 2:
			spotType = VehicleTypeTruck
		}
		level.Spots = append(level.Spots, NewParkingSpot(i, spotType, levelID))
	}
	return level
}

func (pl *ParkingLevel) FindAvailableSpot(vehicleType VehicleType) *ParkingSpot {
	for _, spot := range pl.Spots {
		if !spot.IsOccupied && spot.SpotType == vehicleType {
			return spot
		}
	}
	return nil
}

func (pl *ParkingLevel) FindVehicleSpot(vehicle Vehicle) *ParkingSpot {
	for _, spot := range pl.Spots {
		if spot.Vehicle == vehicle {
			return spot
		}
	}
	return nil
}

File: parking_lot_system/parking_lot/parking_level.py
from .parking_spot import ParkingSpot
from .vehicle import VehicleType

class ParkingLevel:
    def __init__(self, level_id, num_spots):
        self.level_id = level_id
        self.spots = []
        for i in range(num_spots):
            if i % 3 == 0:
                self.spots.append(ParkingSpot(i, VehicleType.CAR, level_id))
            elif i % 3 == 1:
                self.spots.append(ParkingSpot(i, VehicleType.MOTORCYCLE, level_id))
            else:
                self.spots.append(ParkingSpot(i, VehicleType.TRUCK, level_id))

    def find_available_spot(self, vehicle_type):
        for spot in self.spots:
            if not spot.is_occupied and spot.spot_type == vehicle_type:
                return spot
        return None

    def find_vehicle_spot(self, vehicle):
        for spot in self.spots:
            if spot.vehicle == vehicle:
                return spot
        return None
